state/txindex/kv/kv.go

```diff
@@ -613,46 +609,83 @@ LOOP:
 	return filteredHashes
 }
 
-// Keys
+// ##########################  Keys  #############################
+//
+// The indexer has two types of kv stores:
+// 		1. txhash - result
+//    2. event - txhash
+//
+// The event key can be decomposed into 4 parts.
+//    1. A composite key which can be any string.
+//       Usually something like "tx.height" or "account.owner"
+//    2. A value. That corresponds to the key. In the above
+//       example the value could be "5" or "Ivan"
+//    3. The height of the Tx that aligns with the key and value.
+//    4. The index of the Tx that aligns with the key and value
+
+// The event key
+func key(compositeKey, value string, height int64, index uint32) []byte {
+	key, err := orderedcode.Append(
+		nil,
+		compositeKey,
+		value,
+		height,
+		int64(index),
+	)
+	if err != nil {
+		panic(err)
+	}
+	return key
+}
 
-func isTagKey(key []byte) bool {
-	return strings.Count(string(key), tagKeySeparator) == 3
+// parseValueFromKey parses an event key and extracts out the value, returning an error if one arises.
+// This will also involve ensuring that the key has the correct format.
+// CONTRACT: function doesn't check that the prefix is correct. This should have already been done by the iterator
```

Comment:
Why not check it?

Reply:
We only use this function when running a prefixed iterator which means we have already asserted the prefix.

Reply:
Sure, but others may use this function in cases where that isn't true and not be aware of that, so I think it's better to assert this anyway.